code/game/objects/items/food/burgers.dm

```diff
@@ -297,6 +297,13 @@
 	foodtypes = GRAIN | MEAT | DAIRY
 	venue_value = FOOD_PRICE_EXOTIC
 
+/obj/item/food/burger/superbite/suicide_act(mob/user, show_message=TRUE)
+	user.visible_message(span_suicide("[user] starts to eat [src] in one bite, it looks like [user.p_theyre()] trying to commit suicide!"))
+	var/datum/component/edible/component = GetComponent(/datum/component/edible)
+	if(component)
+		component.TakeBite(user, user)
```

Comment:
```suggestion
	component?.TakeBite(user, user)
```

Reply:
Though what does this actually achieve in practice that people are likely to notice? `GetComponent` is a bit touchy.